File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
  int findMin(vector<int> &nums) {
    int l = 0, r = nums.size();
    if (nums[l] <= nums[r - 1]) {
      // Array is not rotated at all.
      return nums[l];
    }

    while (l < r) {
      int mean = (l + r) / 2;
      if (nums[mean] >= nums[0]) {
        l = mean + 1;
      } else {
        r = mean;
      }
    }
    return nums[l];
  }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        res: int = nums[0]
        left: int = 0
        right: int = len(nums) - 1

        while (left <= right):
            if nums[left] < nums[right]:
                res = min(res, nums[left])
                break

            med = ( right + left ) // 2
            res = min (res, nums[med])
            if nums[med] >= nums[left]:
                left = med + 1
            else:
                right = med - 1
        return res
